src/Domain/ElementDistribution.cpp

```diff
@@ -154,7 +154,8 @@ BlockZCurveProcDistribution<Dim>::BlockZCurveProcDistribution(
   size_t element_num_of_block = 0;
   double cost_remaining = total_cost;
   size_t number_of_ignored_procs_so_far = 0;
-  for (size_t i = 0; i < number_of_procs_with_elements; ++i) {
+  // distribute Elements to all but the final proc
+  for (size_t i = 0; i < number_of_procs_with_elements - 1; ++i) {
     size_t global_proc_number = i + number_of_ignored_procs_so_far;
```

Comment:
Optional, but if you define `global_proc_number` outside this loop, then you don't need to recalculate it below. It'll already be at the last proc

Reply:
I can throw this in when I squash!

Reply:
Turns out I actually can't do this and have to recalculate it. This is because when we request a single proc, after the loop, the global proc number represents the last proc, but when we request more than one proc, it will represent the 2nd to last proc. For this reason, I just kept the recalculation